### PR TITLE
[Feature] Trade Changes with NPCs

### PR DESCRIPTION
**Pull request by crdunwel**
https://github.com/EQEmu/Server/issues/932

This should address 1 and 3

### CODE REVIEW

File: zone/trading.cpp [hunk 1]
```diff
@@ -839,6 +835,11 @@ void Client::FinishTrade(Mob* tradingWith, bool finalizer, void* event_entry, st
 						true
 					);
 				}
+                // Can not complete trade with a mob when you are on their hate list
```
Comment (Akkadius) [inline zone/trading.cpp:838]: Indenting is off with the rest of the blocks

**Comment by Akkadius** [comment]
Can anyone validate live intended functionality here? @noudess seems like you'd know this one

**Comment by noudess** [comment]
The not turning to face the trading PC looks good.

Two things about the not competing trade when NPC is aggro:

1) Do we want a GM to be able to do it?  I have no opinion either way.

2) I don't think this PR has that code in the correct place.  Right now I think the code you added is an else case to a large collection of trades that will just happen.

Right now you have:

if (GM | lots of othe cases)
  {
  do trade
  }
elseif (tradingWith->CheckAggro(this))
  {
  dont do trade
  }

They check all cases in the 1st if and explain them all in the elsifs.  You'd have to add your case to the 1st if and leave your code below, or maybe just check and reject up top.

Agree or an I reading it incorrectly?

**Comment by noudess** [comment]
I like this change, and I might be able to fix  #2 in the issue mentioned as well.  Any response to my question above?  I could download a patch and test but I figured you'd either set me straight or make the corrections.

**Comment by Akkadius** [comment]
Bump

**Comment by Akkadius** [comment]
Closing until follow up from contributor

**Pull request closed without merging**